cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java

```diff
@@ -792,13 +801,177 @@ void shouldReturnOkWithOnlyCreatorPersonIdWhenProjectCreatorHasMissingOrganizati
+        assertThat(actualURI, containsString(PARTICIPANT_FACET.getKey() + EQUAL_OPERATOR + "5678"));
+        assertThat(actualURI, containsString(PARTICIPATING_PERSON_ORG_FACET.getKey() + EQUAL_OPERATOR + "194"));
+        assertThat(actualURI, containsString(FUNDING_SOURCE_FACET.getKey() + EQUAL_OPERATOR + "REK"));
+
+        assertEquals(HTTP_OK, response.getStatusCode());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {VERSION_DATE_AGGREGATIONS, VERSION_2023_11_03_AGGREGATIONS, VERSION_NAME_AGGREGATIONS})
+    void shouldAddFacetsToSearchResponse(String acceptHeader) throws Exception {
+        var apiClient = spy(QueryProjectWithFacetsClient.class);
+        var queryResponse = dummyFacetHttpResponse();
+        doReturn(queryResponse).when(apiClient).fetchQueryResults(any());
+        var ignoreEnriched = new HttpResponseFaker(EMPTY_STRING, 404);
+        doReturn(List.of(ignoreEnriched)).when(apiClient).fetchQueryResultsOneByOne(any());
+        doReturn(apiClient).when(clientProvider).getVersionWithFacets();
+        handler = new QueryCristinProjectHandler(clientProvider, new Environment());
+
+        final var queryParams = Map.of("query", "hello",
+                                       "sectorFacet", "UC,INSTITUTE",
+                                       "coordinatingFacet", "185.90.0.0",
+                                       "responsibleFacet", "20754.0.0.0",
+                                       "categoryFacet", "PHD",
+                                       "healthProjectFacet", "CLINICAL",
+                                       "participantFacet", "5678",
+                                       "participantOrgFacet", "194",
+                                       "fundingSourceFacet", "REK");
+
+        var actual = sendQueryWithFacets(queryParams, acceptHeader).getBodyObject(SearchResponse.class);
+
+        assertThat(actual.getAggregations().size(), equalTo(8));
+        assertThat(actual.getHits().size(), equalTo(1));
+    }
+
+    @Test
+    void shouldReturnErrorMessageWithAllFacetParamsMentioned() throws Exception {
+        var input = new HandlerRequestBuilder<Void>(OBJECT_MAPPER)
+                        .withBody(null)
+                        .withQueryParameters(Map.of(INVALID_QUERY_PARAM_KEY, INVALID_QUERY_PARAM_VALUE))
+                        .withHeaders(Map.of(ACCEPT, VERSION_NAME_AGGREGATIONS))
+                        .build();
+
+        handler.handleRequest(input, output, context);
+
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, Problem.class);
+        var body = gatewayResponse.getBodyObject(Problem.class);
+
+        Stream.of(CristinFacetParamKey.values())
+            .dropWhile(hasUnsupportedFacet())
+            .forEach(facetKey -> assertThat(body.getDetail(), containsString(facetKey.getNvaKey())));
+    }
+
+    @Test
+    @Disabled
+    void testtest() throws Exception {
```

Comment:
?

Reply:
removed on next push

Reply:
was a early PR to get early feedback

Reply:
but the branch is almost ready for review and merge, just needing some final adjustments